tests/integration/test_analysisd/test_signature_id/test_signature_id.py

```diff
@@ -0,0 +1,264 @@
+            type: fixture
+            brief: Copies custom rules_file before test, deletes after test.
+        - restart_wazuh_function:
+            type: fixture
+            brief: Restart wazuh at the start of the module to apply configuration.
+
+    assertions:
+        - Check that wazuh starts
+        - Check ".*wazuh-testrule.*Empty 'if_sid' value. Rule '(\\d*)' will be ignored.*"
+
+    input_description:
+        - The `configuration_signature_id_values.yaml` file provides the module configuration for
+          this test.
+        - The `cases_empty_signature_id.yaml` file provides the test cases.
+    '''
+
+    wazuh_log_monitor = FileMonitor(LOG_FILE_PATH)
+
+    # Check logs
+    check_invalid_if_sid(wazuh_log_monitor, metadata['is_empty'])
+
+
+# Tests
+@pytest.mark.tier(level=1)
+@pytest.mark.parametrize('configuration, metadata', zip(t3_configurations, t3_configuration_metadata), ids=t3_case_ids)
+def test_null_signature_id(configuration, metadata, set_wazuh_configuration, truncate_monitored_files,
+                           prepare_custom_rules_file, restart_wazuh_function):
+    '''
+    description: Check that when a rule has an invalid signature ID value, that references a nonexisten rule,
+                 assigned to the if_sid option, the rule is ignored.
+
+    test_phases:
+        - Copy custom rule file into manager
+        - Restart manager
+        - Check logs
+        - Check analysisd is running
+
+    wazuh_min_version: 4.4.0
+
+    tier: 1
+
+    parameters:
+        - configuration:
+            type: dict
+            brief: Configuration loaded from `configuration_template`.
+        - metadata:
+            type: dict
+            brief: Test case metadata.
+        - set_wazuh_configuration:
+            type: fixture
+            brief: Set wazuh configuration.
+        - truncate_monitored_files:
+            type: fixture
+            brief: Truncate all the log files and json alerts files before and after the test execution.
+        - prepare_custom_rules_file:
+            type: fixture
+            brief: Copies custom rules_file before test, deletes after test.
+        - restart_wazuh_function:
+            type: fixture
+            brief: Restart wazuh at the start of the module to apply configuration.
+
+    assertions:
+        - Check that wazuh starts
+        - Check ".*Signature ID '(\\d*)' was not found and will be ignored in the 'if_sid'.* of rule '(\\d*)'" event
+        - Check ".*wazuh-testrule.*Empty 'if_sid' value. Rule '(\\d*)' will be ignored.*"
+
+    input_description:
+        - The `configuration_signature_id_values.yaml` file provides the module configuration for
+          this test.
+        - The `cases_signature_id_values.yaml` file provides the test cases.
+    '''
+
+    wazuh_log_monitor = FileMonitor(LOG_FILE_PATH)
+
+    # Check logs
+    wazuh_log_monitor.start(timeout=T_10, callback=generate_monitoring_callback(CB_SID_NOT_FOUND),
```

Comment:
Create a new event monitor function that wraps this `wazuh_log_monitor.start(...)` call.

Reply:
Done in https://github.com/wazuh/wazuh-qa/pull/3649/commits/0d6fa6f0d7fee02a11d583d081d1b54e639b4678